tests/unit/amplitude/test_parity_prefactor.py

```diff
@@ -81,6 +81,8 @@ def test_parity_prefactor(
         ]
         assert len(in_edge) == 1
         node_id = solution.edges[in_edge[0]].ending_node_id
+        if node_id is None:
+            continue
```

Comment:
What is this good for?

Reply:
To make `mypy` happy. Because `node_id` has `Optional[int]` type prior to line 84.

Reply:
Ok, would asserting also work? That would make the test a bit stronger. So assert node_id is of type int.

Reply:
Not sure, but why not instead of a `continue`?

Reply:
Hmm it should never happen. So if you continue, you would just step over this error if it actually does happen because of some bug.

Reply:
So `assert isinstance(node_id, int)` does not work?